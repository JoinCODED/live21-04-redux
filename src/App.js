import AnimalCage from "./components/AnimalCage";
import { AppWrapper } from "./components/styles";
import ZooAnimals from "./components/ZooAnimals";
import NewAnimalForm from "./components/NewAnimalForm";
import { useState } from "react";

const App = ({ state }) => {
  const [cageAnimal, setCageAnimal] = useState(false);
  const [showNewAnimalForm, setShowNewAnimalForm] = useState(false);

  return (
    <AppWrapper>
      <h2>Welcome to Coded's Zoo!</h2>
      <ZooAnimals
        animals={state.animals}
        setCageAnimal={setCageAnimal}
        setShowNewAnimalForm={setShowNewAnimalForm}
      />
      {cageAnimal ? (
        <AnimalCage setCageAnimal={setCageAnimal} animal={cageAnimal} />
      ) : null}
      {showNewAnimalForm ? (
        <NewAnimalForm
          state={state}
          setShowNewAnimalForm={setShowNewAnimalForm}
        />
      ) : null}
    </AppWrapper>
  );
};

export default App;
