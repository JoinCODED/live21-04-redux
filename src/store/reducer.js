const reducer = () => {
  // Implement the reducer here
};

export default reducer;
