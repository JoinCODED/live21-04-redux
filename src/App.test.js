/**
 *
 *
 *
 *
 *
 *
 *
 *
 *
 * DO NOT MODIFY THIS FILE
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 */

import { render, fireEvent, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import App from "./App";
import reducer from "./store/reducer";
import { createStore } from "redux";

const store = createStore(reducer);

describe("the app", () => {
  let container = null;

  beforeEach(() => {
    const renderedApp = render(
      <>
        <Provider store={store}>
          <App />
        </Provider>
      </>
    );

    container = renderedApp.container;
  });

  it("renders correctly", () => {
    expect(container).toMatchSnapshot();
  });

  it("allows viewing an animal cage by clicking on it", () => {
    fireEvent.click(screen.queryByText("🦍"));

    expect(container).toMatchSnapshot();
  });

  it("allows going back to the zoo after viewing a cage", () => {
    fireEvent.click(screen.queryByText("🦍"));

    expect(screen.queryByText("Back to Zoo")).toBeInTheDocument();
    fireEvent.click(screen.queryByText("Back to Zoo"));

    expect(screen.queryByText("Back to Zoo")).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("allows adding a new animal", () => {
    fireEvent.click(screen.queryByText("➕"));

    expect(screen.queryByText("Back to Zoo")).toBeInTheDocument();
    expect(screen.queryByText("Add Animal")).toBeInTheDocument();

    const nameInput = screen.getByPlaceholderText("Duck");
    const emojiInput = screen.getByPlaceholderText("🦆");

    fireEvent.change(nameInput, { target: { value: "Laila" } });
    fireEvent.change(emojiInput, { target: { value: "🐥" } });

    fireEvent.click(screen.queryByText("Add Animal"));

    expect(screen.queryByText("Back to Zoo")).not.toBeInTheDocument();
    expect(screen.queryByText("Add Animal")).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("allows cancelling adding a new animal", () => {
    fireEvent.click(screen.queryByText("➕"));

    expect(screen.queryByText("Back to Zoo")).toBeInTheDocument();
    expect(screen.queryByText("Add Animal")).toBeInTheDocument();

    const nameInput = screen.getByPlaceholderText("Duck");
    const emojiInput = screen.getByPlaceholderText("🦆");

    fireEvent.change(nameInput, { target: { value: "Khaled" } });
    fireEvent.change(emojiInput, { target: { value: "🥬" } });

    fireEvent.click(screen.queryByText("Back to Zoo"));

    expect(screen.queryByText("Back to Zoo")).not.toBeInTheDocument();
    expect(screen.queryByText("Add Animal")).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
