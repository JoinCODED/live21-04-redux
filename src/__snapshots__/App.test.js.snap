// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the app allows adding a new animal 1`] = `
<div>
  <div
    class="sc-bdfBwQ gnLLvF"
  >
    <h2>
      Welcome to Coded's Zoo!
    </h2>
    <div
      class="sc-gsTCUz hBkqdZ"
    >
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🐅
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Tiger
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🐘
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Elephant
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🦍
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Gorilla
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🐥
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Laila
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          ➕
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Add
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`the app allows cancelling adding a new animal 1`] = `
<div>
  <div
    class="sc-bdfBwQ gnLLvF"
  >
    <h2>
      Welcome to Coded's Zoo!
    </h2>
    <div
      class="sc-gsTCUz hBkqdZ"
    >
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🐅
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Tiger
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🐘
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Elephant
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🦍
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Gorilla
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🐥
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Laila
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          ➕
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Add
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`the app allows going back to the zoo after viewing a cage 1`] = `
<div>
  <div
    class="sc-bdfBwQ gnLLvF"
  >
    <h2>
      Welcome to Coded's Zoo!
    </h2>
    <div
      class="sc-gsTCUz hBkqdZ"
    >
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🐅
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Tiger
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🐘
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Elephant
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🦍
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Gorilla
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          ➕
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Add
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`the app allows viewing an animal cage by clicking on it 1`] = `
<div>
  <div
    class="sc-bdfBwQ gnLLvF"
  >
    <h2>
      Welcome to Coded's Zoo!
    </h2>
    <div
      class="sc-gsTCUz hBkqdZ"
    >
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🐅
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Tiger
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🐘
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Elephant
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🦍
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Gorilla
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          ➕
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Add
        </span>
      </div>
    </div>
    <div
      class="sc-jSgupP djmqUN"
    />
    <div
      class="sc-gKsewC fzKzaE"
    >
      <div
        class="sc-fubCfw hynxmF"
      >
        <h1>
          Gorilla
          s
        </h1>
        <h3>
          🌴   🌴🦍 🦍       🌴🦍
        </h3>
        <button
          class="sc-jrAGrp aYoQV"
        >
          Back to Zoo
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`the app renders correctly 1`] = `
<div>
  <div
    class="sc-bdfBwQ gnLLvF"
  >
    <h2>
      Welcome to Coded's Zoo!
    </h2>
    <div
      class="sc-gsTCUz hBkqdZ"
    >
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🐅
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Tiger
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🐘
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Elephant
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          🦍
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Gorilla
          s
        </span>
      </div>
      <div
        class="sc-dlfnbm kzmQxv"
      >
        <span
          class="sc-hKgILt iLfdF"
        >
          ➕
        </span>
        <span
          class="sc-eCssSg icklDl"
        >
          Add
        </span>
      </div>
    </div>
  </div>
</div>
`;
